profiling-ffi/src/exporter.rs

```diff
@@ -169,6 +163,33 @@ unsafe fn into_vec_files<'a>(slice: Slice<'a, File>) -> Vec<exporter::File<'a>>
         .collect()
 }
 
+#[repr(C)]
+pub enum RequestBuildResult {
+    // This is a Box::into_raw pointer.
+    Ok(*mut Request),
+    Err(ddcommon_ffi::Vec<u8>),
+}
+
+#[cfg(test)]
+impl From<RequestBuildResult> for Result<Box<Request>, String> {
+    fn from(result: RequestBuildResult) -> Self {
+        match result {
+            // Safety: Request is opaque, can only be built from Rust.
+            RequestBuildResult::Ok(ok) => Ok(unsafe { Box::from_raw(ok) }),
+            RequestBuildResult::Err(err) => {
+                // Safety: not generally safe but this is for test code.
+                Err(unsafe { String::from_utf8_unchecked(err.into()) })
+            }
+        }
+    }
+}
+
+/// Drops the result. Since `ddog_prof_Exporter_send` will take ownership of
+/// the Request object, do not call this method if you call
+/// `ddog_prof_Exporter_send` on the `ok` value!
+#[no_mangle]
+pub extern "C" fn ddog_prof_Exporter_Request_BuildResult_drop(_result: RequestBuildResult) {}
+
 /// Builds a Request object based on the profile data supplied.
 ///
 /// # Safety
```

Comment:
Minor: We were discussing it earlier over PM: I think it'd be cool to include here a note that you should either `send` or `drop` the result. I noticed both `drop` and `send` include similar notes, but I think since this is the supplier of the potentially-problematic object, it's cool to also drop a note here :)